eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentEvents.scala

```diff
@@ -144,18 +147,40 @@ object PaymentFailure {
    */
   def transformForUser(failures: Seq[PaymentFailure]): Seq[PaymentFailure] = {
     failures match {
-      case previousFailures :+ LocalFailure(_, RouteNotFound) if previousFailures.nonEmpty => previousFailures
+      case previousFailures :+ LocalFailure(_, _, RouteNotFound) if previousFailures.nonEmpty => previousFailures
       case other => other
     }
   }
 
+  /**
+   * Print a friendly json summary of the payment failure.
+   */
+  def summary(paymentFailed: PaymentFailed): String = {
+    import org.json4s.JsonAST.{JArray, JField, JObject, JString}
+    import org.json4s.{DefaultFormats, jackson}
+    val failures = paymentFailed.failures.map(failure => {
+      val details = failure match {
+        case LocalFailure(_, _, t) => t.getMessage
+        case RemoteFailure(_, _, Sphinx.DecryptedFailurePacket(origin, failureMessage)) => s"$origin returned: ${failureMessage.message}"
+        case _: UnreadableRemoteFailure => "unreadable remote failure"
+      }
+      val route = JArray(failure.route.map(r => JString(r.nextNodeId.value.toHex)).toList)
```

Comment:
We're missing the last hop here:
```suggestion
      val route = JArray((failure.route.map(_.nodeId) ++ failure.route.lastOption.map(_.nextNodeId).toList).map(r => JString(r.nextNodeId.value.toHex)).toList)
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1966/commits/fd216f2abbd0a1163197e7915eca8cccb25c2465